src/framework/theme/components/select/select.component.ts

```diff
@@ -385,7 +381,7 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
    * Selects option or clear all selected options if value is null.
    * */
   protected handleSelect(option: NbOptionComponent<T>) {
-    if (option.value) {
+    if (option.value != null) {
```

Comment:
Could you please swap if and else bodies and use positive comparison (`==` instead `!=`)?
I mean
```
if (option.value == null) {
  this.reset();
} else {
  this.selectOption(option);
}
```
Sorry for bothering you with such changes, but this way it simplifies reading a lot.